### ISSUE TITLE
Highlighting specific sub-fields as invalid when using CustomField + Binder

### ISSUE THREAD

**Issue by vursen** [vaadin/flow-components#4653]
### Describe your motivation

By default, when invalid, CustomField doesn't have any indication e.g. the red background except for having an error message. The recent UX testing showed that it can be difficult to spot an invalid CustomField among other fields in the form, therefore. You can solve that by overriding `setInvalid()` for a CustomField to make it propagate the invalid state to the child fields.

That said, sometimes there is a need to only highlight a specific sub-field as invalid based on the validation error. 

Let's say you have a PhoneField that consists of a country code and phone number sub-fields. When the CustomField value doesn't have a country code, you may expect Binder to show an error message like "Must have a country code" under the whole field but highlight only the country code sub-field as invalid. 

### Describe the solution you'd like

---

**Comment by jcgueriaud1** [comment]
I would probably implement this usecase with an internal binder in the custom field.
So the validation would be displayed properly.
But if you are using an internal binder in the custom field then the custom field will return only a valid value.
So the main binder will always have the valid value.

So you need to create your own validator:
```
binder.forField(phone)
                .asRequired("The field is required")
                // This validator actually checks the country code sub-field.
                .withValidator(
                        value -> phone.checkValidity(value),
                        "Custom field is invalid")
                .bind("user.phone");
```

I'm wondering, as it's an internal behavior of the custom-field, if we should use the `HasValidator` interface for this.
Something like:
```
    public Validator<String> getDefaultValidator() {
        return (value, context) -> {
            return this.checkValidity(value);
        };
    }
```

The documentation seems to describe this as a usecase:
https://github.com/vaadin/flow/blob/main/flow-data/src/main/java/com/vaadin/flow/data/binder/HasValidator.java#L44

In short use the same mechanism for the datetimepicker and a custom field.

**Comment by TatuLund** [comment]
@jcgueriaud1 I have similar experiment here https://github.com/TatuLund/ProtoTools/blob/master/src/main/java/org/vaadin/addons/tatu/prototools/Form.java I.e. sub form as a field.
